### PR TITLE
Update to examples that use Strings for passing arguments.

### PR DESCRIPTION
**Pull request by 2-3-5-41**
tldr:
- Updated "Registering a singleton" example code.
- Changed all relevant examples of `string.into()` -> `&string`.
- Changed "MyEditorSingleton" -> "MyEngineSingleton" in the "engine-singleton.md" page.

### CODE REVIEW

**Comment by Bromeon** [comment]
Thanks, but [as mentioned on Discord](https://discord.com/channels/723850269347283004/1312824986469273621/1312827017233694872), the PR https://github.com/godot-rust/book/pull/62 already handled some of these cases. I merged that PR now, so as a result this one has some conflicts :slightly_smiling_face: 

Also, regarding the new wording in _Builtin types_, I made a suggestion [here](https://github.com/godot-rust/book/pull/62#issuecomment-2508881589), could you apply that verbatim?

I don't think we need to mention v0.2 specifically, there's already a migration guide. Generally we can assume that the book refers to the latest `master` version (maybe with some days delay).

**Comment by Bromeon** [comment]
No need to close, I think there are still some places left, like the _Builtin types_ section :slightly_smiling_face: you can just push to the same branch again and click reopen, let me know if you need any assistance!

File: src/recipes/engine-singleton.md [hunk 1]
```diff
@@ -71,40 +71,42 @@ struct MyExtension;
...
-
-            // Unregistering singleton and freeing the object itself is needed 
-            // to avoid memory leaks and warnings, especially for hot reloading.
-            engine.unregister_singleton(singleton_name);
-            singleton.free();
+            let singleton_name = "MyEngineSingleton";
+
+            // Here, we manually retrieve our singleton(s) that we've registered,
+            // so we can unregister them and free them from memory - unregistering
+            // singletons isn't hanled automaticlly by the library.
```
Comment (Bromeon) [inline src/recipes/engine-singleton.md]:
```suggestion
            // singletons isn't handled automatically by the library.
```

File: src/godot-api/builtins.md [hunk 2]
```diff
@@ -84,8 +84,18 @@ Godot provides three string types: `String` ([`GString`][api-gstring] in Rust),
 `GString` is used as a general-purpose string, while `StringName` is often used for identifiers like class or action names.
 The idea is that `StringName` is cheap to construct and compare.[^string-name-Rust]
 
-These types all support `From` traits to convert to/from Rust `String`, and from `&str`. You can thus use `"some_string".into()` expressions.
-If you need more explicit typing, use `StringName::from("some_string")`.
+When working with Godot APIs, you can pass references to the parameter type (e.g. &GString), as well as Rust strings `&str`, and `&String`.
```
Comment (Bromeon) [inline src/godot-api/builtins.md]:
```suggestion
When working with Godot APIs, you can pass references to the parameter type (e.g. `&GString`), as well as Rust strings `&str`, and `&String`.
```
I missed the backticks there, you should be able to commit directly via web button.

**Review by Bromeon — COMMENTED** [review]
Thanks a lot!

File: src/godot-api/builtins.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ Here is an exhaustive list of all built-in types, by category. We use the GDScri
 - Variant (able to hold anything): `Variant`
 - String types: `String`, `StringName`, `NodePath`
 - Ref-counted containers: `Array` (`Array[T]`), `Dictionary`
-- Packed arrays: `Packed*Array` for following element types:  
+- Packed arrays: `Packed*Array` for following element types:
```
Comment (Bromeon) [inline src/godot-api/builtins.md]: Two empty spaces at the end of a Markdown line mean "line break". Please don't remove them, they're intentionally present for layout reasons :slightly_smiling_face: 

(This is the only occurrence)

Comment (2-3-5-41) [reply src/godot-api/builtins.md]: Sorry, my Zed config auto formats on save, so it did that.

**Comment by Bromeon** [comment]
Thanks a lot! :blush:

**Pull request merged**